zebra/zapi_msg.c

```diff
@@ -2547,6 +2591,71 @@ int zsend_client_close_notify(struct zserv *client, struct zserv *closed_client)
 	return zserv_send_message(client, s);
 }
 
+/* Send response to a srv6 manager connect request to client */
+static void zread_srv6_manager_connect(struct zserv *client,
+				       struct stream *msg, vrf_id_t vrf_id)
+{
+	struct stream *s;
+	uint8_t proto;
+	uint16_t instance;
+	struct vrf *vrf = vrf_lookup_by_id(vrf_id);
+
+	s = msg;
+
+	/* Get data. */
+	STREAM_GETC(s, proto);
+	STREAM_GETW(s, instance);
+
+	/* accept only dynamic routing protocols */
+	if ((proto >= ZEBRA_ROUTE_MAX) || (proto <= ZEBRA_ROUTE_STATIC)) {
+		flog_err(EC_ZEBRA_TM_WRONG_PROTO,
+			 "client %d has wrong protocol %s", client->sock,
+			 zebra_route_string(proto));
+		zsend_srv6_manager_connect_response(client, vrf_id, 1);
+		return;
+	}
+	zlog_notice("client %d with vrf %s(%u) instance %u connected as %s",
+		    client->sock, VRF_LOGNAME(vrf), vrf_id, instance,
+		    zebra_route_string(proto));
+	client->proto = proto;
+	client->instance = instance;
```

Comment:
Aren't these values already set when the zapi session is set up (in the HELLO message processing)? why overwrite them?

Also, the zapi session uniqueness is proto+instance+session_id - you need to account for all three if you're concerned about identifying zapi sessions.

Reply:
Thank you for your comment.
Yes. In zread_hello, proto, instance and session_id is set already.

The reason I added these two lines to the zread_srv6_manager_connect function is because I was referring to the zread_table_manager_connect function and the zread_label_manager_connect function. The first priority was to unify the contents of the functions with these refs. Therefore, I think it is OK to delete this line. 

ref: zread_table_manager_connect
https://github.com/FRRouting/frr/blob/f79d311b3e0ea744cad115978b10bad2661afea9/zebra/zapi_msg.c#L2576-L2577

ref: zread_label_manager_connect
https://github.com/FRRouting/frr/blob/f79d311b3e0ea744cad115978b10bad2661afea9/zebra/zapi_msg.c#L2615-L2617

Our choices are follow:
(0) keep the consistency
(1) zread_srv6_manager_connect: delete these lines
(2) delete these lines on zread_srv6_manager_connect, zread_table_manager_connect and zread_label_manager_connect

Which is your prefer? I'm feeling both is OK. So I want to know your additional comment.

Reply:
I guess I prefer (1): do the right thing in new code, even if there is old code that does something wrong. just as you were tricked by the old code, someone else will be tricked in the future if we don't stop doing the wrong thing?

Reply:
I fixed around these lines.
So this is resolved.
Thank you.